packages/java-shell/src/main/kotlin/com/mongodb/mongosh/MongoShellContext.kt

```diff
@@ -161,13 +165,31 @@ internal class MongoShellContext(client: MongoClient) {
         }
     }
 
-    fun extract(v: Value, type: String? = null): MongoShellResult<*> {
+    fun getShellApiType(rawValue: Value): String? {
+        val rawType = getShellApiTypeFn.execute(rawValue)
+        return if (rawType.isString) rawType.asString() else null
+    }
+
+    fun toShellResult(rawValue: Value): Value {
+        return unwrapPromise(toShellResultFn.execute(rawValue));
+    }
+
+    fun extract(printable: Value?, rawValue: Value? = null): MongoShellResult<*> {
+        var v: Value
+        var type: String? = null
+        if (printable == null) {
+            type = getShellApiType(rawValue as Value)
+            val shellResult = toShellResult(rawValue as Value)
+            v = shellResult["printable"]!!
+        } else {
+            v = printable
+        }
```

Comment:
We can use `val` here
```suggestion
        val v: Value
        val type: String?
        if (printable == null) {
            type = getShellApiType(rawValue as Value)
            v = toShellResult(rawValue)["printable"]!!
        } else {
            type = null
            v = printable
        }
```